sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,300 @@
+```
+#### New
+```C#
+string vmName = "quickstartvm";
+AvailabilitySetData aSetData = new AvailabilitySetData(location);
+AvailabilitySet aset = (await resourceGroup.GetAvailabilitySets().CreateOrUpdateAsync(vmName + "_aSet", aSetData)).Value;
+string asetId = aset.Id;
+```
+
+Parameters can be specified via the `AvailabilitySetData` object, in here, the basic default only requires the location. The availability set is created using  the `GetAvailabilitySets()` extension method instead of using another client. 
+
+### Create a Virtual Network and Subnet
+#### Old
+```C#
+string vnetName = vmName + "_vnet";
+string subnetName = "mySubnet";
+
+VirtualNetwork vnet = new VirtualNetwork()
+{
+    Location = location,
+    AddressSpace = new AddressSpace()
+    {
+        AddressPrefixes = new List<string>()
+                {
+                    "10.0.0.0/16",
+                }
+    },
+    Subnets = new List<Subnet>()
+            {
+                new Subnet()
+                {
+                    Name = subnetName,
+                    AddressPrefix = "10.0.0.0/24",
+                }
+            }
+};
+VirtualNetwork putVnetResponse = networkClient.VirtualNetworks.CreateOrUpdate(rgName, vnetName, vnet);
+VirtualNetwork subnetResponse = networkClient.Subnets.Get(rgName, vnetName, subnetName);
+```
+#### New
+```C#
+string vnetName = "MYVM" + "_vnet";
+string subnetName = "mySubnet";
+string[] addressPrefixes = { "10.0.0.0/16" };
+AddressSpace addressSpace = new AddressSpace(addressPrefixes);
+
+VirtualNetworkData vnetData = new VirtualNetworkData()
+{
+    AddressSpace = addressSpace,
+    Subnets =
+    {
+        new SubnetData()
+        {
+            Name = subnetName,
+            AddressPrefix = "10.0.0.0/24"
+        }
+    }
+};
+VirtualNetwork vnet = (await resourceGroup.GetVirtualNetworks().CreateOrUpdateAsync(vnetName, vnetData)).Value;
+```
+
+The main difference here is that a virtual network object is no longer needed to create a virtual network. One similarity is that subnets are defined inside virtual networks, however, with the new SDK you can get a subnets container using `.GetSubnets()`, and from there create any subnet in the virtual network from which the method is being called.
+
+### Create a Security Group
+#### Old
+```C#
+string nsgName = vmName + "_nsg";
+NetworkSecurityGroup nsgParameters = new NetworkSecurityGroup()
+{
+    Location = location
+};
+
+NetworkSecurityGroup putNSgResponse = networkClient.NetworkSecurityGroups.CreateOrUpdate(rgName, nsgName, nsgParameters);
+NetworkSecurityGroup nsg = networkClient.NetworkSecurityGroups.Get(rgName, nsgName);
+```
+#### New
+```C#
+string nsgName = vmName + "_nsg";
+NetworkSecurityGroupData nsgData = new NetworkSecurityGroupData() { Location = location };
+_ = resourceGroup.GetNetworkSecurityGroups().CreateOrUpdateAsync(nsgName, nsgData);
+```
+
+Creating a Network security group does not longer require a `NetworkSecurityGroup` object as a parameter.
```

Comment:
```suggestion
Creating a Network security group no longer requires a `NetworkSecurityGroup` object as a parameter.
```

Reply:
I think I would just remove this line - actually the structure of the models is very similar between the SDKs, the main difference is in operation signatures, and how the operations them selves are accessed.